src/app/components/nonce-setter.tsx

```diff
@@ -8,18 +8,21 @@ import { useNextNonce } from '@app/query/stacks/nonce/account-nonces.hooks';
 
 interface NonceSetterProps {
   children: ReactNode;
+  skipNonceSetting?: boolean;
 }
-export function NonceSetter({ children }: NonceSetterProps) {
+export function NonceSetter({ children, skipNonceSetting }: NonceSetterProps) {
```

Comment:
Perhaps we could just conditionally not render this component? 

`!skipNonceSetting && <NonceSetter />`?